tests/dind.sh

```diff
@@ -1,15 +1,31 @@
-#!/bin/sh
+#!/bin/bash
 
+K3D_EXE=${EXE:-/bin/k3d}
 K3D_IMAGE_TAG=$1
 
+# define E2E_KEEP to non-empty for keeping the e2e runner container after running the tests
+E2E_KEEP=${E2E_KEEP:-}
+
+####################################################################################
+
+TIMESTAMP=$(date "+%H%M%S")
+
 k3de2e=$(docker run -d --rm \
           -v "$(pwd)"/tests:/tests \
           --privileged \
-          -e EXE="/bin/k3d" \
+          -e EXE="$K3D_EXE" \
           -e CI="true" \
+          --name "k3d-e2e-runner-$TIMESTAMP" \
           k3d:$K3D_IMAGE_TAG)
 
 sleep 5 # wait 5 seconds for docker to start
 
 # Execute tests
-docker exec $k3de2e /tests/runner.sh
\ No newline at end of file
+finish() {
+  if [ -z "$E2E_KEEP" ] ; then
+    docker stop "$k3de2e" || /bin/true
```

Comment:
Ah sorry, what I meant here is, that we can actually `docker rm -f` the container even, if `$E2E_KEEP` is not set instead of only stopping it. We should probably stop it in any case and rm it in case of failure.

Reply:
Aha, ok. We are currently starting the container with `--rm`, so I will remove that flag and, in the `finish()`, 1) stop the container 2) remove it unless `E2E_KEEP` is set. I think that's what you mean, right?

Reply:
Ah true, I forgot about the `--rm`.
Yes, that's what I meant, but I'm not sure if that's ideal or if we should always remove except if `E2E_KEEP` was set.. :thinking: What do you think?

Reply:
I think we should always stop the container, and then keep it only if `E2E_KEEP` is set (and remove it otherwise). As you said, maybe we could be interested in keeping the container...

Reply:
> I think we should always stop the container, and then keep it only if `E2E_KEEP` is set (and remove it otherwise). As you said, maybe we could be interested in keeping the container...

Always only `stop` or also `remove`?

Reply:
Always stop and remove. But, if E2E_KEEP is set, only stop.